### PR TITLE
Allow reordering of schemes in list

### PR DESCRIPTION
**Pull request by Pete-Y-CS**
Currently have logic lifted to ListMode because it feels like that's what should be managing the order of schemes but I don't mind putting it in PerSchemeControls if we prefer

### CODE REVIEW

File: src/lib/sidebar/ListMode.svelte [hunk 1]
```diff
@@ -55,6 +105,15 @@
 <hr />
 
 {#each Object.keys($gjSchemeCollection.schemes) as scheme_reference}
-  <PerSchemeControls {scheme_reference} />
+  <PerSchemeControls {scheme_reference}>
+    {#if Object.keys($gjSchemeCollection.schemes).length > 1}
+      <SecondaryButton on:click={moveSchemeInList(scheme_reference, -1)}>
```
Comment (dabreegster) [inline src/lib/sidebar/ListMode.svelte:110]:
```suggestion
      <SecondaryButton on:click={() => moveSchemeInList(scheme_reference, -1)}>
```
then you don't need to return a callback from the function

File: src/lib/sidebar/ListMode.svelte [hunk 2]
```diff
@@ -40,6 +40,56 @@
...
+          newSchemeOrder,
+          $gjSchemeCollection.schemes
+        );
+        $gjSchemeCollection.schemes = newSchemesObject;
+        $gjSchemeCollection = $gjSchemeCollection;
+      }
+    };
+  }
+
+  function swapArrayValues(
```
Comment (dabreegster) [inline src/lib/sidebar/ListMode.svelte:69]: Optional: https://www.freecodecamp.org/news/swap-two-array-elements-in-javascript/ "How to Swap Two Array Elements by Destructuring" looks a bit nicer to me, just a one-liner

File: src/lib/sidebar/ListMode.svelte [hunk 3]
```diff
@@ -40,6 +40,56 @@
...
+  ) {
+    const firstValue = array[firstIndex];
+    const secondValue = array[secondIndex];
+    array[secondIndex] = firstValue;
+    array[firstIndex] = secondValue;
+
+    return array;
+  }
+
+  function getReorderedSchemesObject(
```
Comment (dabreegster) [inline src/lib/sidebar/ListMode.svelte:82]: All of this works fine for me in modern Firefox, but according to MDN, `Object.keys` does not guarantee any kind of ordering. https://stackoverflow.com/questions/5525795/does-javascript-guarantee-object-property-order is a minefield; I couldn't find a canonical newer answer (preferably in someplace like MDN). We should maaaybe use something like https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyNames instead, but it seems to work now, so let's just go with it.

**Review by dabreegster — APPROVED** [review]
Nice and simple!

File: src/lib/sidebar/ListMode.svelte (hunk 1 above)
Comment (Pete-Y-CS) [reply src/lib/sidebar/ListMode.svelte:110]: I don't think this works?

Comment (Pete-Y-CS) [reply src/lib/sidebar/ListMode.svelte:110]: Or at least I don't understand it, so for now I'm gonna leave this change

File: src/lib/sidebar/ListMode.svelte (hunk 2 above)
Comment (Pete-Y-CS) [reply src/lib/sidebar/ListMode.svelte:69]: done, just testing

**Pull request merged**